src/CallbackResolver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use Invoker\CallableResolver;
+use Invoker\Exception\NotCallableException;
+
+/**
+ * This alternative resolver uses the generic Invoker to support PHP-DI's extended callable syntax
+ */
+class CallbackResolver extends \Silex\CallbackResolver
+{
+    /**
+     * @var CallableResolver
+     */
+    private $resolver;
+
+    public function __construct(\Pimple $app, CallableResolver $resolver)
+    {
+        $this->resolver = $resolver;
+        parent::__construct($app);
+    }
+
+    /**
+     * @param string $name
+     * @return array|callable
+     */
+    public function convertCallback($name)
+    {
+        // original pattern
+        if ($this->isValid($name)) {
+            return parent::convertCallback($name);
+        }
+
+        // try to resolve callback from container
+        try {
+            return $this->resolver->resolve($name);
+        } catch (NotCallableException $e) {
+            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $name));
+        }
+    }
+
+    /**
+     * @param string|callable $name
+     * @return callable
+     */
+    public function resolveCallback($name)
+    {
+        // return immediately if $name is callable
+        if (is_callable($name)) {
+            return $name;
+        }
```

Comment:
I'm afraid this might be an issue as method calls are considered as statically callable. For example `['Logger', 'warning']` is considered callable by PHP even though it's *not* a static method, so calling `Logger::warning()` would be wrong and the "correct" callable would be `[$logger, 'warning']`. Have a look at this line: https://github.com/PHP-DI/Invoker/blob/master/src/CallableResolver.php#L69

Anyway all that to say that this shortcut will create issues with some types of callables.